### PR TITLE
CI: build with OpenBLAS

### PR DESCRIPTION
**Pull request by xoviat**
Closes #3.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -20,7 +20,6 @@ environment:
       TEST_MODE: fast
       APPVEYOR_SAVE_CACHE_ON_ERROR: true
       APPVEYOR_SKIP_FINALIZE_ON_EXIT: true
-      BUILD_COMMIT: v1.13.0rc2
```
Comment (charris) [inline appveyor.yml]: Why are you removing that?

Comment (xoviat) [reply appveyor.yml]: It's not removed; it's down below (from merge conflicts). I can reset it if you want.

**Comment by xoviat** [comment]
@charris Ping.

**Comment by charris** [comment]
Restarted the stalled test. There is one warning issued during the tests, I'm not sure it is relevant
```
clang: warning: -Wl,-rpath=/var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/tmpg6wkjc4r/libbar.so: 'linker' input unused
```

**Comment by charris** [comment]
The use of mingw and cygwin could use some explanation. You could add an explanatory section at the beginning explaining in outline how things are done. Are both mingw and cygwin guaranteed to be available?

@matthew-brett @pv Thoughts?

**Comment by pv** [comment]
It's essentially the same procedure we use for Scipy.

Note that Cygwin is *not* used (the commands only clear up pip cache).

**Comment by charris** [comment]
OK, let's give this a shot.

@matthew-brett Do you know if the problem with OpenBLAS Cholesky on 32 bit windows is fixed?

**Comment by charris** [comment]
Thanks @xoviat .

**Comment by pv** [comment]
The openblas binaries used here are compiled with vanilla mingw-w64, not mingwpy, and do not have the issue with cholesky.

**Comment by charris** [comment]
>  do not have the issue with cholesky.

Good to know, thanks for the info.

**Comment by njsmith** [comment]
Does that mean that they mess with the fp status word? Do they rely on
that? We've seen weird issues before either because loading a mingw-w64 dll
causes extended precision to be enabled for other routines that weren't
expecting it, or else that the mingw-w64 routines misbehave when called
from a thread, because the thread doesn't have extended precision enabled.
(Sorry if this is way off base.)

**Comment by xoviat** [comment]
I have not seen mingw change the fpu mode. I *have* seen MSVC change the fpu mode on 32 bits.

**Comment by xoviat** [comment]
Also, to clarify, when I say "mingw," I am never talking about mingw.org.

**Comment by njsmith** [comment]
mingw definitely changes the fpu mode to enable "extended" precision by
default; the windows standard is to use only double precision.

**Comment by xoviat** [comment]
All I know is that numpy is built with this exact configuration and there are no FPU mode changes found by numpy/numpy#9574. That suggests to me that MinGW does not change the FPU mode, at least in this case.

**Comment by njsmith** [comment]
That test doesn't do anything until after numpy is imported, at which point
it has already changed the fpu mode.

**Comment by xoviat** [comment]
Assuming that the FPU mode is changed, the question is what to do about it. For Python >= 3.5, I think I'll have an OpenBLAS ready that doesn't use MinGW, but is fully compatible with the CRT that can go into the next numpy release.

For older Pythons, the options (that I can think of) are:
1. Accept horrible performance (maybe go back to ATLAS)
2. Live with the FPU mode change

One option that I would not recommend is switching back to mingwpy, as the project has been abandoned.

**Comment by pv** [comment]
I don't think this results to fpu mode changes on import, at least based on a quick check.

**Pull request merged**